crates/revm/src/evm_impl.rs

```diff
@@ -326,6 +413,47 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
                     );
                 }
 
+                #[cfg(feature = "optimism")]
+                if self.data.env.cfg.optimism && !is_deposit {
+                    // If the transaction is not a deposit transaction, fees are paid out
+                    // to both the Base Fee Vault as well as the L1 Fee Vault.
+
+                    // Fetch the L1 block information from account storage.
+                    let Ok(l1_block_info) = optimism::fetch_l1_block_info(&mut self.data.db) else {
+                        panic!("Failed to fetch L1 block info from account storage.");
+                    };
+
+                    // Calculate the L1 cost of the transaction based on the L1 block info.
+                    // TODO(clabby): This is incorrect. The L1 cost is computed with the full
+                    // enveloped encoding of the transaction, not its input. How do we get the full
+                    // encoded tx here? We may pass it through the `TxEnv`, but that feels hacky.
```

Comment:
Maybe pass the `l1_cost` into the `TxEnv` directly? Reduces the size of the diff and moves the L1 cost computation out of revm, but not sure if I like this.